src/tools/arrayFilters.ts

```diff
@@ -0,0 +1,17 @@
+/**
+ * Spots "positional operator" path in string for array filters
+ * Example: xxx.$[element_name].yyy
+ *
+ * @see https://www.mongodb.com/docs/manual/reference/operator/update/positional-filtered/
+ */
+const ARRAY_FILTER_OPERATION_PATTERN = /\$\[(\w+)]/g;
+
+/**
+ * Checks the path contains at least one positional operator (ie ".$[element]")
+ *
+ * @param path Path with potentially a positional operator "my.$[element_for_array_filter].key"
+ * @returns true if the input path needs an arrayFilter option when using a $unset on it.
+ */
+export function isArrayFilterPath(path: string) {
+  return !!path.match(ARRAY_FILTER_OPERATION_PATTERN);
```

Comment:
@hugop95 We can't because this is  a regex with global flag
The behavior is to return `true` until there are no match left, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/test#using_test_on_a_regex_with_the_global_flag
```ts
> const ARRAY_FILTER_OPERATION_PATTERN = /\$\[(\w+)]/g;
> ARRAY_FILTER_OPERATION_PATTERN.test("$[str]a")
true
> ARRAY_FILTER_OPERATION_PATTERN.test("$[str]a")
false
> ARRAY_FILTER_OPERATION_PATTERN.test("$[str]ab")
true
> ARRAY_FILTER_OPERATION_PATTERN.test("$[str]abc")
false
```

When it's not global it's as you'd expect
```ts
> const ARRAY_FILTER_OPERATION_PATTERN_not_global = /\$\[(\w+)]/;
undefined
> ARRAY_FILTER_OPERATION_PATTERN_not_global.test("$[str]")
true
> ARRAY_FILTER_OPERATION_PATTERN_not_global.test("$[str]")
true
```

Reply:
@pp0rtal I did not see that! 😯
Thanks for the explanation!